### DISCUSSION TITLE
Data Binding with Mount Method

### DISCUSSION THREAD

**Discussion by tmgomas** [livewire/livewire#3963] [1. Help]
I Guys 

I am new to livewire and  i am trying develop app .. 

and i use one component to collect some fees in my student 

My case is  I have 3 Payment option like monthly Termly and yearly  

so i use select option to select that option like below code 

```
<Select wire:model="check">"
    <option value="1">Monthly</option>
    <option value="2">Termly</option>
    <option value="3">Yearly</option>
</Select>
```

it normally happen in normal method but i need to use mount method

```
public function mount($studentsData)
    {
        $studentsData = $this->studentsData;
        $this->fc_student_id = $studentsData[0]->students_id;
        $this->fc_fee_master_id = $studentsData[0]->fee_masters_id;

if($this->fc_fee_scheem_id == 3 && $StudentCFBalace = DB::table('school_fee_collectors')->latest('fc_id')->where('school_fee_collectors.fc_student_id',$studentsData[0]->students_id)->first()){ 
            $this->fc_amount = $studentsData[0]->ifs_amount - 500 - $StudentCFBalace->fc_cf_amount ;
        }
        elseif($this->fc_fee_scheem_id == 3){
            $this->fc_amount = $studentsData[0]->ifs_amount - 500;
        }
        elseif($StudentCFBalace = DB::table('school_fee_collectors')->latest('fc_id')->where('school_fee_collectors.fc_student_id',$studentsData[0]->students_id)->first())
        {
            $this->fc_amount = $studentsData[0]->ifs_amount - $StudentCFBalace->fc_cf_amount ; 
        }
        else{
            $this->fc_amount = $studentsData[0]->ifs_amount;
        }
       
        
    }
```

when i use mount method that value not passing to if condition .. what i do

**Comment by denizozturk** [comment]
Hello,

I'm reviewing your code, but this structure has some inconsistencies.
You have already squashed the `$studentsData` variable in the mount function in the `$studentsData = $this->studentsData;` code. If you are going to send the values of the variable to the livewire component the code should be `$this->studentsData = $studentsData;`.

If you could describe the problem/error you're encountering more clearly, I might be able to help.
